source/val/validate_mode_setting.cpp

```diff
@@ -347,6 +349,12 @@ spv_result_t ValidateExecutionMode(ValidationState_t& _,
       case spv::ExecutionMode::LocalSizeId:
       case spv::ExecutionMode::FPFastMathDefault:
       case spv::ExecutionMode::MaximumRegistersIdINTEL:
+      case spv::ExecutionMode::IsApiEntryAMDX:
+      case spv::ExecutionMode::MaxNodeRecursionAMDX:
+      case spv::ExecutionMode::MaxNumWorkgroupsAMDX:
+      case spv::ExecutionMode::ShaderIndexAMDX:
+      case spv::ExecutionMode::SharesInputWithAMDX:
+      case spv::ExecutionMode::StaticNumWorkgroupsAMDX:
```

Comment:
Again simple tests for the new execution modes being accepted or not by the appropriate version of OpExecutionMode.

Reply:
Test added.